File: #Notion/Code/3. Sort/bubble.c
#include <stdio.h>

int array[10] = {1, 10, 6, 4, 8, 2, 9, 3, 7, 5};

void printArray(int data[], int n) {
    int i;

    for (i = 0; i < n; i++)
        printf("%d ", data[i]);
    printf("\n");
}

int main(void) {
    int i, j, temp;
    int n = sizeof(array) / sizeof(array[0]);

    printArray(array, n);

    for (i = 0; i < n; i++) {
        for (j = 0; j < n - 1 - i; j++) {
            if (array[j] > array[j + 1]) {
                temp = array[j];
                array[j] = array[j + 1];
                array[j + 1] = temp;
            }
        }
    }

    printArray(array, n);

    return 0;
}

File: #Notion/Code/3. Sort/Bubble.java
public class Bubble {
    private static int[] data = { 8, 4, 3, 7, 9, 5, 1 };

    private static void bubbleSort(int[] data, int n) {
        int temp;

        for (int i = n - 1; i > 0; i--) {
            for (int j = 0; j < i; j++) {
                if (data[j] > data[j + 1]) {
                    temp = data[j];
                    data[j] = data[j + 1];
                    data[j + 1] = temp;
                }
            }
        }
    }

    private static void printArray(int[] data) {
        for (int d : data)
            System.out.print(d + " ");
        System.out.println();
    }

    public static void main(String[] args) {
        printArray(data);
        bubbleSort(data, data.length);
        printArray(data);
    }
    
}